src/components/link-component/LinkComponent.tsx

```diff
@@ -0,0 +1,48 @@
+import { FC } from 'react'
+
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import LinkOutlinedIcon from '@mui/icons-material/LinkOutlined'
+
+import AppButton from '~/components/app-button/AppButton'
+
+import { ButtonVariantEnum, Link } from '~/types'
+
+import { styles } from '~/components/link-component/LinkComponent.styles'
+import { spliceSx } from '~/utils/helper-functions'
+
+interface LinkComponentProps {
+  link: Link
+}
+
+const openLink = (link: Link) => {
+  window.open(link.url, '_blank', 'noopener noreferrer')?.focus()
+}
+
+const LinkComponent: FC<LinkComponentProps> = ({ link }) => {
+  return (
+    <Box sx={styles.linkWrapper}>
+      <AppButton
+        onClick={() => {
+          openLink(link)
+        }}
+        sx={styles.linkButton}
+        variant={ButtonVariantEnum.Text}
+      >
+        <Box sx={styles.formatFrame}>
+          <LinkOutlinedIcon sx={styles.linkIcon} />
+        </Box>
+        <Box sx={styles.linkInfo}>
+          <Typography sx={spliceSx(styles.secondaryText, styles.sizeLimit)}>
+            {link.name}
+          </Typography>
+          <Typography sx={spliceSx(styles.link, styles.sizeLimit)}>
```

Comment:
you can use sx={ styles.link } only here and in styles create an object with sizeLimit styles, and then spread it in those two styles:

const sizeLimit=  {
 width: '92%',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textAlign: 'start'
}

export const styles = {
  link: {
    typography: TypographyVariantEnum.Caption,
    color: 'primary.500',
    textDecoration: 'underline',
    ...sizeLimit
  }
}

Reply:
> you can use sx={ styles.link } only here and in styles create an object with sizeLimit styles, and then spread it in those two styles:
> 
> const sizeLimit= { width: '92%', textOverflow: 'ellipsis', whiteSpace: 'nowrap', overflow: 'hidden', textAlign: 'start' }
> 
> export const styles = { link: { typography: TypographyVariantEnum.Caption, color: 'primary.500', textDecoration: 'underline', ...sizeLimit } }

thank you for your advice, fixed